english_tasks/irregular_verbs.py

```diff
@@ -74,52 +102,81 @@ def display_statistic(list1, list2, list3):
-                print(current_verb)
-            next_try = input('Do you want to do it again? (yes/no): ')
-            while next_try != 'no':
-                user_answer = input('Enter here: ')
-                user_verbs = user_answer.split()
-                result = verbs_comparing(user_verbs, current_verb[1:])
-                if result:
-                    fixed_verbs.append(current_verb[0])
-                    wrong_verbs.remove(current_verb[0])
-                    break
-                else:
-                    next_try = input('Do you want to do it again? (yes/no): ')
-
-        else:
-            right_verbs.append(current_verb[0])
-        list_of_rows.remove(current_verb)
-    ask_stat = input('Do you want to see a statistic?(yes/no): ')
-    if ask_stat == 'yes':
-        display_statistic(right_verbs, wrong_verbs, fixed_verbs)
+    arguments = get_args()
 
+    # read the special file to store verbs into a list
+    verbs_list = rows_list(arguments.verbs_filename)
+    greeting()
+    name = input('Enter your name: ')
+    nb_verb = int(input('Enter the number of verbs: '))
 
-if __name__ == '__main__':
-    main()
+    if not 0 < nb_verb <= len(verbs_list):
+        print('Error! You entered a wrong value (max number is 74)')
+        sys.exit(1)
 
-# подсказка на первый раз каждый раз
-# только yes и no
-# добавить перевод
+    for verb in range(1, nb_verb + 1):
+        print(f'\nverb №{verb}')
 
+        # generate and display a random verb from the list and display the V1 of it
+        current_verb = random_verb(verbs_list)
+        print(current_verb[0])
+        verb_v1 = current_verb[0]
+
+        # read the answers from the user and compare it with correct v2 and v3 forms
+        answer_v2 = input('Enter the V2: ')
+        answer_v3 = input('Enter the V3: ')
+        result_v2 = compare_verb(answer_v2, current_verb[1])
+        result_v3 = compare_verb(answer_v3, current_verb[2])
+
+        # if all answers are correct, display it, save the verb in the list
+        # and continue executing with next verb, otherwise give the user
+        # 3 attempts to fix it
+        if result_v2 and result_v3:
+            print('All form is correct!\n')
+            RESULT_DICT['Correct Verbs'].append(verb_v1)
+            continue
+
+        # check if the v2 is incorrect
+        if not result_v2:
+            print('\nV2 is wrong!')
+            if not generate_attempts(2, current_verb[1], 2):
+                print(current_verb[1])
+
+        # check if the v3 is incorrect
+        if not result_v3:
+            print('\nV3 is wrong!')
+            if not generate_attempts(2, current_verb[2], 3):
+                print(current_verb[2])
+
+        verbs_list.remove(current_verb)
+
+    # display all statistic and store the result into a file
+    ask_stat_save = input('Do you want to save your result(y/n)?:')
+    if ask_stat_save.upper() == 'y':
+        display_statistic(name, nb_verb, RESULT_DICT)
```

Comment:
do not ask a user for this, use optional ArgumentParser parameter `-o` which contains the path to the file that user specifies